fdbrpc/sim2.actor.cpp

```diff
@@ -870,7 +870,15 @@ class Sim2 : public ISimulator, public INetworkConnections {
 		return emptyConfig;
 	}
 
-	virtual void stop() { isStopped = true; }
+	virtual void stop() {
+		isStopped = true;
+		for ( auto& fn : stopCallbacks ) {
+			fn();
+		}
```

Comment:
I think this doesn't pose the same level of risk as with the net2 run loop because the sim2 run loop can't run any new tasks after this is called. I suppose it could continue running the current task, but I don't know if that's really a concern, and I assume this doesn't cause any correctness problems in testing. Is that your understanding?

Reply:
It doesn't cause problems, because we don't use EIO in simulation.   But I see your point, and should adjust this as well.